src/js/outreach/views/verify_views.js

```diff
@@ -0,0 +1,129 @@
+  ui: {
+    submit: '.js-submit',
+  },
+  triggers: {
+    'click @ui.submit': 'click:submit',
+  },
+  modelEvents: {
+    'change:hasError': 'render',
+  },
+  template: hbs`
+    <div class="verify__icon verify__icon--blue">{{fat "user-lock"}}</div>
+    <h2 class="verify__heading-text">Request a verification code to view this health resource.</h2>
+    <p class="verify__info-text">We’ll send a text message with a verification code to the phone number 123-456-7890.</p>
+    <button class="verify__submit button--green w-100 js-submit">Send Verification Code</button>
+  `,
+  onChangeDate() {
+    const dob = this.ui.date.val();
+    this.model.set({ dob });
+    this.ui.submit.prop('disabled', !dob);
+  },
+  onBlurDate() {
+    this.model.set({ hasError: false });
+  },
+  onClickSubmit() {
+    this.ui.submit.prop('disabled', true);
+  },
+});
+
+const VerifyCodeView = View.extend({
+  ui: {
+    submit: '.js-submit',
+    resend: '.js-resend',
+  },
+  triggers: {
+    'click @ui.submit': 'click:submit',
+    'click @ui.resend': 'click:resend',
+  },
+  template: hbs`
+    <div class="verify__icon verify__icon--blue">{{fat "user-lock"}}</div>
+    <h2 class="verify__heading-text">Enter your verification code.</h2>
+    <p class="verify__info-text">We sent a text message with a verification code to the phone number 123-456-7890.</p>
+    <div class="verify__code-fields">
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+    </div>
+    <button class="verify__submit button--green w-100 js-submit" disabled>Send Verification Code</button>
+    <div class="verify__heading-text u-text-link js-resend">Send a new code</div>
+  `,
+  onRender() {
+    const inputElements = this.$('.js-input');
+
+    inputElements.each((index, element) => {
+      element.addEventListener('keydown', event => {
+        if (event.keyCode === 8 && event.target.value === '') {
+          inputElements[Math.max(0, index - 1)].focus();
+        }
+      });
+
+      element.addEventListener('input', event => {
+        const [first, ...rest] = event.target.value;
+
+        event.target.value = first ?? '';
+
+        const isLastInputEl = index === inputElements.length - 1;
+        const didInsertContent = first !== undefined;
+
+        const code = inputElements.map((i, el) => el.value).get().join('');
+
+        if (code.length === inputElements.length) {
+          this.enableSubmitButton();
+        } else {
+          this.disableSubmitButton();
+        }
+
+        if (didInsertContent && !isLastInputEl) {
+          inputElements[index + 1].focus();
+          inputElements[index + 1].value = rest.join('');
+          inputElements[index + 1].dispatchEvent(new Event('input'));
+        }
+      });
+    });
+  },
```

Comment:
When one input is entered, it will jump to the next one. And it also allows a code to be copy-pasted and you can backspace between the inputs.

Screen recording of the input functionality: 

https://github.com/RoundingWell/care-ops-frontend/assets/35355575/039cc118-d04b-48cf-a89c-f7930f3ad079

I couldn't get this to work via the Marionette.js `triggers: {}`, so need to investigate that more.

Reply:
Just updated this to use marionette.js to handle the `input` and `keydown` input events.